src/posix/client.cpp

```diff
@@ -182,7 +182,7 @@ static bool DoWrite(int aFile, const void *aBuffer, size_t aSize)
         else
         {
             aBuffer = reinterpret_cast<const uint8_t *>(aBuffer) + rval;
-            aSize -= rval;
+            aSize -= (size_t)rval;
```

Comment:
Prefer C++-style casts:
```suggestion
            aSize -= static_cast<size_t>(rval);
```